domenetjenester/inngangsvilkar/src/main/java/no/nav/foreldrepenger/inngangsvilkaar/medlemskap/v2/MedlemskapsvilkårRegel.java

```diff
@@ -0,0 +1,119 @@
+package no.nav.foreldrepenger.inngangsvilkaar.medlemskap.v2;
+
+import static no.nav.foreldrepenger.inngangsvilkaar.medlemskap.v2.MedlemskapAksjonspunktÅrsak.BOSATT;
+import static no.nav.foreldrepenger.inngangsvilkaar.medlemskap.v2.MedlemskapAksjonspunktÅrsak.MEDLEMSKAPSPERIODER_FRA_REGISTER;
+import static no.nav.foreldrepenger.inngangsvilkaar.medlemskap.v2.MedlemskapAksjonspunktÅrsak.OPPHOLD;
+import static no.nav.foreldrepenger.inngangsvilkaar.medlemskap.v2.MedlemskapAksjonspunktÅrsak.OPPHOLDSRETT;
+import static no.nav.foreldrepenger.inngangsvilkaar.medlemskap.v2.MedlemskapsvilkårRegelGrunnlag.Personopplysninger.PersonstatusPeriode.*;
+import static no.nav.fpsak.tidsserie.LocalDateTimeline.JoinStyle;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import no.nav.foreldrepenger.inngangsvilkaar.medlemskap.v2.MedlemskapsvilkårRegelGrunnlag.Personopplysninger;
+import no.nav.fpsak.tidsserie.LocalDateSegment;
+import no.nav.fpsak.tidsserie.LocalDateTimeline;
+
+//TODO reimplementeres i fp-inngangsvilkår
+final class MedlemskapsvilkårRegel {
+
+    private MedlemskapsvilkårRegel() {
+    }
+
+    static Set<MedlemskapAksjonspunktÅrsak> kjørRegler(MedlemskapsvilkårRegelGrunnlag grunnlag) {
+        var resultat = new HashSet<MedlemskapAksjonspunktÅrsak>();
+        utledMedlemskapPerioderÅrsak(grunnlag).ifPresent(resultat::add);
+        utledBosattÅrsak(grunnlag).ifPresent(resultat::add);
+        utledOppholdÅrsak(grunnlag).ifPresent(resultat::add);
```

Comment:
LovligOpphold. Se også siste kommentarer på arbeidssiden / avklaringer.
Her og neste punkt kan være aktuelt å vurdere for periodepå 1 dag (STP), -12m og +inntil3år